File: bbc.py
import database
import requests
from bs4 import BeautifulSoup
import re


def scraping_bbc_pages():
    res = requests.get("https://www.bbc.com/news")
    soup = BeautifulSoup(res.text, "html.parser")
    grn = soup.find_all("a")
    stitle = soup.find_all(class_="gs-c-promo-heading gs-o-faux-block-link__overlay-link gel-pica-bold "
                                  "nw-o-link-split__anchor")
    ftitle = soup.find_all(class_="gs-c-promo-heading gs-o-faux-block-link__overlay-link gel-paragon-bold "
                                  "nw-o-link-split__anchor")
    title = ftitle + stitle
    summary = soup.find_all(class_="gs-c-promo-summary gel-long-primer gs-u-mt nw-c-promo-summary")
    date = soup.find_all(class_="gs-o-bullet__text date qa-status-date gs-u-align-middle gs-u-display-inline")
    area = soup.find_all(class_="gs-c-section-link gs-c-section-link--truncate nw-c-section-link nw-o-link "
                                "nw-o-link--no-visited-state")
    news_list = (filtering_bbc_news(title, summary, date, area))

    title = []
    titlelink = []
    time = []
    location = []
    detail = []
    for news in news_list:
        title.append(news["title"].replace("'", '"'))
        titlelink.append(news['title_link'].replace("'", '"'))
        time.append(news['time'].replace("'", '"'))
        location.append(news['location'].replace("'", '"'))
        detail.append(news['detail'].replace("'", '"'))
    database.insert_into_database("bbc",title, titlelink, time, location, detail)

    return "bbc database updated"


def filtering_bbc_news(title, summary, date, area):
    news_list = []
    for index, item in enumerate(title[:8]):
        title = item.getText()
        url = "https://www.bbc.com" + item.get("href")
        time = re.split("h|m|day",date[index].getText(),1)[1]
        detail = summary[index].getText()
        location = area[index].getText()

        match_found = 0
        for tit in news_list:
            header = tit["title"]
            if header == title:
                match_found = 1
                continue
        if match_found == 0:
            current_news = {'title': title, 'title_link': url, 'time': time,
                            'location': location, 'detail': detail}
            news_list.append(current_news)

    return news_list[:5]

File: frontend/src/components/bbc/Bbc.js
import React from "react";
// import './Navbar.css';

const Bbc = ({ newsList }) => {
    // console.log(newsList);
    return(
        <div className="bg-brown">
            <h2>BBC</h2>
           <div>
               <a href={newsList.titlelink[0]}>{newsList.title[0]}</a>
            </div>
            <div>
                <p>{newsList.detail[0]}</p><br/>
                <p>{newsList.location[0]} {newsList.time[0]}</p>
            </div>
            <div>
               <a href={newsList.titlelink[1]}>{newsList.title[1]}</a>
            </div>
            <div>
                <p>{newsList.detail[1]}</p><br/>
                <p>{newsList.location[1]} {newsList.time[1]}</p>
            </div>
            <div>
               <a href={newsList.titlelink[2]}>{newsList.title[2]}</a>
            </div>
            <div>
                <p>{newsList.detail[2]}</p><br/>
                <p>{newsList.location[2]} {newsList.time[2]}</p>
            </div>
            <div>
               <a href={newsList.titlelink[3]}>{newsList.title[3]}</a>
            </div>
            <div>
                <p>{newsList.detail[3]}</p><br/>
                <p>{newsList.location[3]} {newsList.time[3]}</p>
            </div>
            <div>
               <a href={newsList.titlelink[4]}>{newsList.title[4]}</a>
            </div>
            <div>
                <p>{newsList.detail[4]}</p><br/>
                <p>{newsList.location[4]} {newsList.time[4]}</p>
            </div>
        </div>
    )
}

export default Bbc;